examples/routing_service/routing_service_file_adapter/cpp/FileInputDiscoveryStreamReader.cxx

```diff
@@ -0,0 +1,65 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include "FileInputDiscoveryStreamReader.hpp"
+
+using namespace rti::routing;
+using namespace rti::community::examples;
+
+const std::string FileInputDiscoveryStreamReader::SQUARE_FILE_NAME = "Input_Square.csv";
+const std::string FileInputDiscoveryStreamReader::CIRCLE_FILE_NAME = "Input_Circle.csv";
+const std::string FileInputDiscoveryStreamReader::TRIANGLE_FILE_NAME = "Input_Triangle.csv";
+
+bool FileInputDiscoveryStreamReader::fexists(const std::string filename)
+{
+    std::ifstream input_file;
+    input_file.open(filename);
+    return input_file.is_open();
+}
+
+FileInputDiscoveryStreamReader::FileInputDiscoveryStreamReader(const PropertySet &)
+{
+    /**
+     * In our example, we provide statically the stream information available. 
+     * Make sure, the stream name and the stream type matches the <stream_name> 
+     * and <registered_type_name> respectively in the XML configuration file.
+     */
+    if (fexists(SQUARE_FILE_NAME)) {
+        this->data_samples_.push_back(new StreamInfo("Square", "ShapeType"));
+    }
+
+    if (fexists(CIRCLE_FILE_NAME)) {
+        this->data_samples_.push_back(new StreamInfo("Circle", "ShapeType"));
+    }
+
+    if (fexists(TRIANGLE_FILE_NAME)) {
+        this->data_samples_.push_back(new StreamInfo("Triangle", "ShapeType"));
+    }
+}
+
+void FileInputDiscoveryStreamReader::dispose()
```

Comment:
This operation should receive a stream name as an argument, since streams can be disposed independently

Reply:
Also this class should be the one holding the listener, and notifiying when appropriate

Reply:
Agreed. I have changed it to receive a pointer to StreamInfo.

Reply:
I have also moved the listener inside this class.